src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.cpp

```diff
@@ -43,11 +43,136 @@ MemRefType convertToMemRefType(Type type) {
   return memRefType;
 }
 
+/// Retrieve function which contains the current operation.
+FuncOp getContainingFunction(Operation *op) {
+  Operation *parentFuncOp = op->getParentOp();
+
+  // While parent is not a FuncOp and its cast to a FuncOp is null.
+  while (!llvm::dyn_cast_or_null<FuncOp>(parentFuncOp))
+    parentFuncOp = parentFuncOp->getParentOp();
+
+  return cast<FuncOp>(parentFuncOp);
+}
+
+void addInitBlock(PatternRewriter &rewriter, Location loc, FuncOp function) {
+  // If this is the first time we encounter an operation in this
+  // function, we create an entry inside the initMap and split the
+  // function body into an init block and a main block.
+  //
+  // function func_name() {
+  //    ... init block ...
+  //    br ^bb1
+  //  ^bb1:  // pred: ^bb0
+  //    ... main block ...
+  //    return
+  // }
+  //
+  // Note: the block ^bb0 being the first block has its label omitted.
+  //
+  if (initMap.count(function) == 0) {
+    ONNXOperandsInitState *initState = new ONNXOperandsInitState();
```

Comment:
Same here, we should avoid new by making use of smart pointers.

Reply:
Done.